deploy.ps1

```diff
@@ -10,8 +10,31 @@ param (
     [string[]] $WebRedirectUris,
     [string[]] $CorsUrls,
     [string] $TenantId,
-    [string] $SubscriptionId
+    [string] $SubscriptionId,
+    [ValidateSet('Global', 'USGov', 'USGovDoD')]
+    [string] $Environment="Global"
 )
+
+# Define the mapping between Connect-MgGraph environments and Get-AzAccount environments
+$environmentMap = @{
+    "Global"   = "AzureCloud"
+    "USGov"    = "AzureUSGovernment"
+    "USGovDoD" = "AzureUSGovernment"  # Both map to the same Azure US Government cloud for Azure operations
+}
+
+# Function to map the Graph environment to the Azure environment
+function Get-AzureEnvironmentFromGraph {
```

Comment:
I think you can remove this function and access the value from the hash table directly on line 53. If it doesn't find it in the hash table it will throw an error.